src/PropertyList/PropertyList.tsx

```diff
@@ -0,0 +1,73 @@
+import React, { ReactNode } from 'react';
+import Immutable from 'immutable';
+import { AttributeGrid, AttributeList, Caption, Content } from './propertlyListStyles';
+import { ThemeProvider } from '@emotion/react';
+import theme from 'src/styles/theme';
+
+export interface Property {
+    /** This is to set the caption for the PropertyList */
+    caption: string;
+    /** This is to set the caption style that needs to be added to the caption */
+    captionStyles?: string | null;
+    /** This is to set the content for the corresponding caption */
+    content: (data: any) => ReactNode;
+    /** This is to set the style for the content associated to the caption */
+    contentStyles?: string | null;
+}
+
+interface BaseProps {
+    /** This is to set the list of the property- caption, captionStyle, content and contentStyles */
+    attributes: Immutable.List<Property>;
```

Comment:
I thought that we were moving away from `Immutable`?  If that's still the case, this should be updated.